File: Lista Prof. Renan/Exercicios Lista 1/C/André/NumVog.cpp
//André Fernandes Gonçalves
//Matrícula 114038073

#include <stdio.h> //printf
#include <conio.h> //getch
#include <locale.h> //setlocale
#include <string.h> //fgets


int main(){
	setlocale(LC_ALL, "Portuguese");
	char frase[20];
	int numvog=0, i=0;
	
	printf("\n O programa conta o número de vogais em uma frase. \n");
	printf("\n Insira a frase: \n");
	fgets(frase,20,stdin);
	
	while (frase[i] != '\0'){
		if (frase[i] == 'a'|| frase[i] == 'e'|| frase[i] == 'i'|| frase[i] == 'o'|| frase[i] == 'u'){
			numvog++;
		}
		else if (frase[i] == 'A'|| frase[i] == 'E'|| frase[i] == 'I'|| frase[i] == 'O'|| frase[i] == 'U'){
			numvog++;
		}
		else if (frase[i] == 'Á'|| frase[i] == 'á'|| frase[i] == 'Â'|| frase[i] == 'â'|| frase[i] == 'Ã'|| frase[i] == 'ã'){
			numvog++;
		}
		else if (frase[i] == 'É'|| frase[i] == 'é'|| frase[i] == 'Ó'|| frase[i] == 'ó'|| frase[i] == 'Ô'|| frase[i] == 'ô'){
			numvog++;	
		}
		i++;
	}
	printf("\n O número de vogais é: %d \n", numvog);
	getch();
	return(0);
}

File: Lista Prof. Renan/Exercicios Lista 1/FORTRAN/NumVog.f90
!====================================================================
!André Fernandes Gonçalves
!Matrícula 114038073
!Linguagens de Programação para Engenharia Elétrica
!====================================================================
PROGRAM NumVog
IMPLICIT NONE
!====================================================================
!Entrada de dados

INTEGER vogais
CHARACTER palavra*30
PRINT*,"Conta o número de vogais em sua palavra"
PRINT*," "
PRINT*,"Insira a sua palavra"
READ(*,*)palavra
!====================================================================
!Processamento

vogais=0
DO i=1,LEN_TRIM(palavra)
    SELECT CASE(palavra(i:i))
    CASE('A','E','I','O','U','a','e','i','o','u')
        vogais=vogais+1
    END SELECT
END DO
!====================================================================
!Saída de dados
PRINT*," "
PRINT*,"O numero de vogais",vogais

!====================================================================
STOP
END
